mlir/include/mlir/Dialect/Vector/IR/VectorOps.td

```diff
@@ -1363,7 +1363,7 @@ def Vector_TransferReadOp :
                    AffineMapAttr:$permutation_map,
                    AnyType:$padding,
                    Optional<VectorOf<[I1]>>:$mask,
-                   OptionalAttr<BoolArrayAttr>:$in_bounds)>,
+                   BoolArrayAttr:$in_bounds)>,
```

Comment:
I think we could minimize the impact in tests if we use `DefaultValuedAttr` to set inbound dims to `false` and avoid printing that case. That should "mostly" preserve the existing behavior while fixing the optionality ambiguity.

Reply:
It's possible to set the default value to an empty array (`{}`), e.g. :
```mlir
DefaultValuedAttr<BoolArrayAttr, "{}">:$in_bounds
```

Sadly, that's not sufficient.

Reply:
I believe here we'd want a `DefaultValuedAttr<DenseBoolArrayAttr, "{ false, false }">` (if that syntax works?)

Reply:
Sure, but that will only work when the underlying vector is rank 2. How could we make it work for arbitrary ranks?

Reply:
Ah I see, we just need custom code then: provide a builder which does not take this argument and construct the array based on the rank?

Reply:
I was able to fix that by updating the parsers. The suggestion with builder was very helpful - put me on the right path, thanks!

Reply:
Nice! Is the printer also handling the default?

Reply:
Yes, that's already the case: https://github.com/llvm/llvm-project/blob/1ed84a862f9ce3c60251968f23a5405f06458975/mlir/lib/Dialect/Vector/IR/VectorOps.cpp#L3974-L3976